### PR TITLE
[AGNTR-180] Update build_links task to consider patch_release flag

### PR DESCRIPTION
**Pull request by kacper-murzyn**
### What does this PR do?

`update_build_links` is run as part of the Agent Release Candidate build pipeline. In case of working on minor release and a patch release at the same time, there is no distinction between those two within the task and the [QA build links page](https://datadoghq.atlassian.net/wiki/spaces/agent/pages/2889876360/Build+links+-+7.53.0-rc.2) gets updated regardless, which is not what we want, as this page is meant for minor release QA.

I left the option to create patch release links just in case it's needed.

### Motivation

Making Agent Release process easier.

### Describe how to test/QA your changes

Tested locally as much as possible. No further verification is neeed.

### CODE REVIEW

File: tasks/release.py [hunk 1]
```diff
@@ -1621,11 +1625,11 @@ def update_build_links(_ctx, new_version):
 
     BUILD_LINKS_PAGE_ID = 2889876360
 
-    match = RC_VERSION_RE.match(new_version)
+    match = RC_VERSION_RE.match(new_version) if patch_version else MINOR_RC_VERSION_RE.match(new_version)
     if not match:
         raise Exit(
             color_message(
-                f"{new_version} is not a valid Agent RC version number/tag. \nCorrect example: 7.50.0-rc.1",
+                f"{new_version} is not a valid Agent RC minor version number/tag or this is a path release. \nCorrect example: 7.50.0-rc.1",
```
Comment (chouetz) [inline tasks/release.py:1632]:
```suggestion
                f"{new_version} is not a valid {'patch' if patch else 'minor'} Agent RC version number/tag.\nCorrect example: 7.50{'.0' if patch else ''}-rc.1",
```

**Review by chouetz — COMMENTED** [review]
Minor comments we should probably ship, wdyt?

File: tasks/release.py [hunk 2]
```diff
@@ -1605,11 +1608,12 @@ def _check_commit_in_repo(repo_name, branch, release_json_field):
...
 
 @task
-def update_build_links(_ctx, new_version):
+def update_build_links(_ctx, new_version, patch_version=False):
     """
     Updates Agent release candidates build links on https://datadoghq.atlassian.net/wiki/spaces/agent/pages/2889876360/Build+links
 
-    new_version - should be given as an Agent 7 RC version, ie. '7.50.0-rc.1' format.
+    new_version - should be given as an Agent 7 RC minor version, ie. '7.50.0-rc.1' format.
+    patch-version - if set to True, then task can be used for patch releases, ie. '7.50.1-rc.1' format. Otherwise patch release number will be considered as invalid.
```
[the remark this answers is not in the document]
Comment (kacper-murzyn) [reply tasks/release.py:1616]: I like those, let's add it!

File: tasks/release.py [hunk 3]
```diff
@@ -40,6 +40,9 @@
 # Regex matching rc version tag format like 7.50.0-rc.1
 RC_VERSION_RE = re.compile(r'\d+[.]\d+[.]\d+-rc\.\d+')
 
+# Regex matching minor release rc version tag (no patch version) like 7.50-rc.1, but not 7.50.1-rc.1
```
Comment (KevinFairise2) [inline tasks/release.py:43]: I think the regex will match `7.50.0-rc.1` and not `7.50.1-rc.1`. But that's not exactly what the comment says, `7.50-rc.1` is not matched by the regex

Comment (KevinFairise2) [reply tasks/release.py:43]: This regex: `\d+[.]\d+-rc\.\d+` would do what the comment says

Comment (kacper-murzyn) [reply tasks/release.py]: Right, thats for spotting that. The issue was actually in the comment, not the regex.

Comment (KevinFairise2) [reply tasks/release.py]: I think that now you have an issue with the other suggestions that Nicolas made

Comment (kacper-murzyn) [reply tasks/release.py]: I checked the descriptions once more, they should be accurate now

File: tasks/release.py [hunk 4]
```diff
@@ -40,6 +40,9 @@
 # Regex matching rc version tag format like 7.50.0-rc.1
 RC_VERSION_RE = re.compile(r'\d+[.]\d+[.]\d+-rc\.\d+')
 
+# Regex matching minor release rc version tag (no patch version) like 7.50.0-rc.1, but not 7.50.1-rc.1
```
Comment (chouetz) [inline tasks/release.py:43]:
```suggestion
# Regex matching minor release rc version tag like x.y.0-rc.1 (semver PATCH == 0), but not x.y.1-rc.1 (semver PATCH > 0)
```
If I understand correctly a `minor release rc version`. I just want to highlight the difference with [semver](https://semver.org/) definitions :)

**Review by chouetz — APPROVED** [review]

**Pull request merged**